compiler/luci/import/src/PostImport.cpp

```diff
@@ -231,18 +241,20 @@ class ValidateNodeProp final : public luci::CircleNodeMutableVisitor<void>
 
       auto cond_graph_input = cond_graph_inputs->at(cond_in->index());
       auto body_graph_input = body_graph_inputs->at(body_in->index());
-      if ((cond_in->rank() != body_in->rank()))
+      if (cond_graph_input->shape()->rank() != body_graph_input->shape()->rank())
```

Comment:
Why do you need to change this? `cond_in` and `cond_graph_input` is different.

Reply:
AFAIK, `PostImport.cpp` is for checking the graph itself after all subgraphs are imported.
https://github.com/Samsung/ONE/blob/38f825ae5e8d0cf2a90f6528e246699c89988b37/compiler/luci/import/src/Importer.cpp#L289

Therefore, checking shape/dtype of `GraphInputContext` seems correct in `PostImport.cpp`.
Checking shape/dtype of `CircleNode` will be done at shape/type inference and `Validate.cpp`.

In addition, current codes are not consistent because shape/dtype of both `GraphInputContext` and `CircleNode` are used without any criteria.

Reply:
Validation here is to check the shape/dtype of the nodes and graph; related subgraphs for `IF` and `WHILE`.
I don't actually understand what you wrote, thus I can't disgree or find any problems with it.
If you want to change, please add some comments as you understand and what you are checking and the conditions you think should be.